direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java

```diff
@@ -1632,6 +1633,25 @@ public boolean onError(Throwable error) {
                 .collect(Collectors.summingLong(o -> ((TopicOffset) o).getOffset())));
   }
 
+  @Test
+  public void testObserveOnNonExistingTopic() throws InterruptedException {
+    Accessor accessor = kafka.createAccessor(direct, entity, storageUri, partitionsCfg(3));
+    LocalKafkaLogReader reader = accessor.newReader(context());
+    try {
+      // need this to initialize the consumer
+      assertNotNull(reader.getPartitions());
+      reader.validateTopic(reader.getConsumer(), "non-existing-topic");
```

Comment:
how does this test that the validateTopic method gets invoked at the right moment?

Reply:
It does not. The problem is that there is no way to inject "wrong" topic to the test (mock) reader.